### ISSUE TITLE
Spewy warning "UserWarning: Please use quant_min and quant_max to specify the range for observers."

### ISSUE THREAD

**Issue by silvasean** [pytorch/pytorch#61054]
## 🐛 Bug

I am getting a UserWarning through no apparent fault of my own from the following program: [code](https://github.com/llvm/mlir-npcomp/blob/79928cd2dd6e61573f62aecf47b967b727eee824/frontends/pytorch/e2e_testing/torchscript/quantized_models.py#L1)

The core is simply:
```
def get_quantized_mlp():
    model = QuantizedMLP()
    model.eval()
    model.qconfig = torch.quantization.default_qconfig
    torch.quantization.prepare(model, inplace=True)
    torch.manual_seed(0)
    for _ in range(32):
        model(get_mlp_input())
    torch.quantization.convert(model, inplace=True)
    return model
```

I believe that this program is a pretty vanilla use of the torch quantization API's, so it would be nice if it didn't warn me on something that I'm not responsible for.

## To Reproduce

1. Set up https://github.com/llvm/mlir-npcomp (see README)
1. Run TorchScript e2e tests.

(happy to provide more detail, but I'm hoping that this is understandable by inspection)

The specific error message is:
```
/usr/local/google/home/silvasean/.local/lib/python3.9/site-packages/torch/quantization/observer.py:135: UserWarning: Please use quant_min and quant_max to specify the range for observers.                     reduce_range will be deprecated in a future release of PyTorch.
  warnings.warn(
```

## Expected behavior

I expect to see no warnings.

## Environment

```
Collecting environment information...
PyTorch version: 1.10.0.dev20210628+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux rodete (x86_64)
GCC version: (Debian 10.2.1-6+build2) 10.2.1 20210110
Clang version: 11.0.1-2
CMake version: version 3.18.4
Libc version: glibc-2.31

Python version: 3.9.2 (default, Feb 28 2021, 17:03:44)  [GCC 10.2.1 20210110] (64-bit runtime)
Python platform: Linux-5.10.28-1rodete2-amd64-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: Quadro P5000
Nvidia driver version: 460.56
cuDNN version: /usr/local/cuda-10.1/targets/x86_64-linux/lib/libcudnn.so.7.6.4
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.10.0.dev20210628+cpu
[pip3] torchvision==0.11.0.dev20210628+cpu
[conda] _pytorch_select           0.2                       gpu_0  
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h6bb024c_0  
[conda] mkl                       2020.1                      217  
[conda] mkl-include               2020.2                      256  
[conda] mkl-service               2.3.0            py38he904b0f_0  
[conda] mkl_fft                   1.1.0            py38h23d657b_0  
[conda] mkl_random                1.1.1            py38h0573a6f_0  
[conda] numpy                     1.18.5           py38ha1c710e_0  
[conda] numpy-base                1.18.5           py38hde5b4d6_0  
[conda] numpydoc                  1.1.0                      py_0  
[conda] pytorch                   1.6.0           py3.8_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] torchvision               0.7.0                py38_cu101    pytorch
```

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

**Comment by jerryzh168** [comment]
why didn't we land Angela's PR?

**Comment by vkuzo** [comment]
it would be great to have some context on why the warning is there originally (why are we planning to deprecate reduce_range), that would help understanding whether removing it makes sense or not

**Comment by jerryzh168** [comment]
I see, I'm not in initial discussion, but my guess is `reduce_range` can be fully expressed with `quant_min` and `quant_max` in observers and it is a bit redundant to have this flag in observers.

**Comment by jerryzh168** [comment]
warning is added in this PR: https://github.com/pytorch/pytorch/pull/42602, maybe @raghuramank100 or @z-a-f would have more context

**Comment by vkuzo** [comment]
> I see, I'm not in initial discussion, but my guess is `reduce_range` can be fully expressed with `quant_min` and `quant_max` in observers and it is a bit redundant to have this flag in observers.

makes sense, `reduce_range` seems easier to use for non-experts though.  Is the code cleanliness benefit we get from removing this warning worth the usability hit?

**Comment by jerryzh168** [comment]
I feel it's probably not worth it right now since this is the only flag we have and fbgemm is the major backend people use, but considering we might have different options coming up, like with xnnpack, the range is [-127, 127], or lower bit quantizations, where user need to set the range manually as well, or gpu quantization, I feel it will be worth it as we support more backends since fbgemm will become less "special" at that time and we probably don't want to introduce flags for each backend.

**Comment by vkuzo** [comment]
I see, then landing https://github.com/pytorch/pytorch/pull/61419 sounds good, do you want to commandeer it or do you need me to?

**Comment by jerryzh168** [comment]
sounds good, this can be a ramp up task for new team members, I have listed this as a task for Terry, just want to confirm with you that we actually want to do it

**Comment by vkuzo** [comment]
ok, let me pass the issue over to you then

**Comment by terrychenism** [comment]
fixed by PR #71027
